config_ext_test.go

```diff
@@ -0,0 +1,461 @@
+		run(t, TestCase{
+			Value:     provider.Get("occupants").Get("honda").Get("driver"),
+			HasValue:  true,
+			Interface: "jane",
+			Populate:  &s,
+		})
+		assert.Equal(t, "jane", s, "unexpected result after populate")
+	})
+
+	t.Run("default missing value", func(t *testing.T) {
+		var s string
+		val, err := provider.Get("not_there").WithDefault("something")
+		require.NoError(t, err, "couldn't set default")
+		run(t, TestCase{
+			Value:     val,
+			HasValue:  true,
+			Interface: "something",
+			Populate:  &s,
+		})
+		assert.Equal(t, "something", s, "unexpected result after populate")
+	})
+
+	t.Run("default overriden by scalar", func(t *testing.T) {
+		var s string
+		val, err := provider.Get("practical.honda").WithDefault("CRV")
+		require.NoError(t, err, "couldn't set default")
+		run(t, TestCase{
+			Value:     val,
+			HasValue:  true,
+			Interface: "civic",
+			Populate:  &s,
+		})
+		assert.Equal(t, "civic", s, "unexpected result after populate")
+	})
+
+	t.Run("default overriden by nil", func(t *testing.T) {
+		t.Skip("TODO: panics")
+		var s string
+		val, err := provider.Get("nothing").WithDefault("something")
+		require.NoError(t, err, "couldn't set default")
+		run(t, TestCase{
+			Value:     val,
+			HasValue:  true,
+			Interface: nil,
+			Populate:  &s,
+		})
+		assert.Equal(t, "", s, "unexpected result after populate")
+	})
+
+	t.Run("default merges maps", func(t *testing.T) {
+		t.Skip("TODO: defaults aren't deep-merged")
+		var m map[string]string
+		val, err := provider.Get("practical").WithDefault(map[string]string{
+			"ford":   "fiesta",
+			"toyota": "corolla",
+		})
+		require.NoError(t, err, "couldn't set default")
+		run(t, TestCase{
+			Value:    val,
+			HasValue: true,
+			Interface: map[interface{}]interface{}{
+				"honda":  "civic",
+				"toyota": "corolla",
+				"nissan": "altima",
+				"ford":   "fiesta",
+			},
+			Populate: &m,
+		})
+		assert.Equal(t, map[string]string{
+			"honda":  "civic",
+			"toyota": "corolla",
+			"nissan": "altima",
+			"ford":   "fiesta",
+		}, m, "unexpected result after populate")
+	})
+
+	t.Run("default doesn't merge sequences", func(t *testing.T) {
+		var s []string
+		val, err := provider.Get("fun").WithDefault([]string{"delorean"})
+		require.NoError(t, err, "couldn't set default")
+		run(t, TestCase{
+			Value:     val,
+			HasValue:  true,
+			Interface: []interface{}{"maserati", "lamborghini"},
+			Populate:  &s,
+		})
+		assert.Equal(t, []string{"maserati", "lamborghini"}, s, "unexpected result after populate")
+	})
+
+	t.Run("default ignores type mismatches", func(t *testing.T) {
+		// TODO: We should return an error in strict mode.
+		// fun is a sequence, so we shouldn't be able to merge in a mapping
+		val, err := provider.Get("fun").WithDefault(map[string]string{"dodge": "viper"})
+		assert.NoError(t, err, "error on type mismatch in permissive mode")
+		assert.Equal(t, []interface{}{"maserati", "lamborghini"}, val.Value(), "wrong merged output")
+	})
```

Comment:
This is wrong; defaults should act like a `base_base.yaml`, which controls the type of each key. The end result here should be `{dodge: viper}`.